wd_core.au3

```diff
@@ -1302,7 +1303,13 @@ EndFunc   ;==>_WD_Startup
 ; Link ..........:
 ; Example .......: No
 ; ===============================================================================================================================
-Func _WD_Shutdown($vDriver = Default)
+Func _WD_Shutdown($vDriver = Default, $iDelay = Default)
+	If $iDelay = Default Then $iDelay = 2000
+	If $iDelay Then __WD_Sleep($iDelay)
```

Comment:
To be honest @Danp2, this confused me. I tested it and of course it worked, but this expression is kind of unclean 😔 .

`If $iDelay Then ...`
I would expect a boolean which is checked for true or false, not "is variable defined".

`__WD_Sleep(($iDelay = Default) ? 2000 : $iDelay)`
This would be enough or do you want to catch the case of invalid parameters too?

I hope it's understandable what I mean?

Best regards
Sven

Reply:
Yes, I get your point. My intention was to allow the Sleep to be bypassed if the value of $iDelay was zero. 

> __WD_Sleep(($iDelay = Default) ? 2000 : $iDelay)
This would be enough or do you want to catch the case of invalid parameters too?

This is a good solution. However, it doesn't match the Default parameter handling used throughout the rest of the UDF. What do you think about this instead?

```autoit
	If $iDelay = Default Then $iDelay = 2000
	If IsInt($iDelay) And $iDelay > 0 Then __WD_Sleep($iDelay)
```

P.S. Thanks for testing and providing feedback!

Reply:
> My intention was to allow the Sleep to be bypassed if the value of $iDelay was zero.

Understood 👍 .


``` autoit
    If $iDelay = Default Then $iDelay = 2000
    If IsInt($iDelay) And $iDelay > 0 Then __WD_Sleep($iDelay)
```

That's nice and clean and out of my perspective 😀 . Well done Dan.